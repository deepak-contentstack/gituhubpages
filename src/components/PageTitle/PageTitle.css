@import '../../../tokens/variables.scss';
@import '../../../tokens/tokens.css';
@import '../../static/styles/css/typography.css';
@import '../../static/styles/css/variable.css';
@import '../../static/styles/css/common.css';

.PageTitle {
    color: $color-brand-secondary-dark;
    font-family: $font-family-primary;
    font-size: 1.28rem;
    font-weight: $font-weight-extra-bold;
    line-height: 1.6875rem;
    
    &.PageTitle--editable{
        line-height: $space-30;
        padding: 0 $space-5;
        &:hover{
            background-color: $color-brand-secondary-lighter;
            cursor: pointer;
        }
    }

    .PageTitle--ct_name {
        border: 1px solid $color-border-light;
        border-radius: $radii-4;
        color: $color-font-secondary;
        font-size: $size-font-tiny;
        font-weight: $font-weight-medium;
        letter-spacing: -6e-05em;
        margin-left: $space-8;
        padding: $space-4;
        text-decoration: none;
    }
}
