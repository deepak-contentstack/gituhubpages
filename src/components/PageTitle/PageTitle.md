## **PageTitle**

It's the title of the page, commonly used in pageHeader component

#### **Neccessary Props**:

- **title**: String
    -Page title name