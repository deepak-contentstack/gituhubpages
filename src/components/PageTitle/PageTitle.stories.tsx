import React from "react";
import { storiesOf } from "@storybook/react";
import { withKnobs, text, boolean, number } from "@storybook/addon-knobs";
import { withA11y } from "@storybook/addon-a11y";
import { PageTitle } from "../../index";
import notes from "./PageTitle.md";

export default {
  title: "Components / PageTitle",
  component: PageTitle,
  decorators: [withKnobs, withA11y],
  parameters: {
    notes: { notes },
    propTypes: PageTitle,
  },
};

export const Default = () => (
  <div>
    <PageTitle title={text("title", "Page Title 1")} ctdetails={() => {
      return <span className="PageTitle--ct_name"> CT Title </span>
    }}></PageTitle>
  </div>
);
