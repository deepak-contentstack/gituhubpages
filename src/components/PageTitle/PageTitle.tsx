import React from 'react'
import './PageTitle.css'
import cn from 'classnames'

class PageTitle extends React.Component<any, any> {
  render() {
    const { className, title, ctdetails, ...otherProps } = this.props

    const classNames = cn('PageTitle', className)

    return (
      <div className={classNames} {...otherProps}>
        {title}
        {ctdetails && ctdetails()}
        {/* {ctdetails && ctdetails.title ? <span className="PageTitle--ct_name"> {ctdetails.title} </span> : null} */}
      </div>
    )
  }
}

export default PageTitle
