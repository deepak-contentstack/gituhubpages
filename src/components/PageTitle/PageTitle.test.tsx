import React from 'react'
import renderer from 'react-test-renderer'

import PageTitle from './PageTitle'

describe('PageTitle', () => {
  it('render PageTitle', () => {
    const DefaultPageTitle = () => {
      return <PageTitle title="Page Title 1"></PageTitle>
    }
    const tree = renderer.create(<DefaultPageTitle />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
