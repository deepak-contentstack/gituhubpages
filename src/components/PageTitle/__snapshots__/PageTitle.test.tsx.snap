// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle render PageTitle 1`] = `
<div
  className="PageTitle"
>
  Page Title 1
</div>
`;
